on-chain/contracts/Games.sol

```diff
@@ -0,0 +1,41 @@
+pragma solidity ^0.4.18;
+
+contract Games {
+    
+    // game instance information
+    struct GameInstance {
+        string game_name;
+        string encrypted_result;
+    }
+    
+    mapping (address => GameInstance) gameInstances;// mapping of instance address with its details
```

Comment:
instead of struct directlly mapping can be used
mapping(address => bytes32) contests;